GitUI/CommandsDialogs/FormPush.cs

```diff
@@ -504,13 +504,14 @@ private bool HandlePushOnExit(ref bool isError, FormProcess form)
                 {
                     if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force"))
                     {
-                        if (GitCommandHelpers.VersionInUse.SupportPushForceWithLease)
+                        if(form.ProcessArguments.StartsWith("push ", StringComparison.OrdinalIgnoreCase))
```

Comment:
Agree with your proposals and remarks. Push new version.
Though I use Trace.Assert instead of Debug.Assert, so this condition will be verified even in Release.